test/open_api/map_test.exs

```diff
@@ -0,0 +1,403 @@
+             } = componentCallback
+
+      assert %{
+               "NotFound" => %OpenApiSpex.Response{
+                 headers: %{
+                   "X-Rate-Limit-Limit": %OpenApiSpex.Header{
+                     description: "The number of allowed requests in the current period",
+                     schema: %OpenApiSpex.Schema{
+                       type: :integer
+                     }
+                   }
+                 },
+                 content: %{
+                   "application/json": %OpenApiSpex.MediaType{
+                     schema: %OpenApiSpex.Schema{}
+                   }
+                 },
+                 links: %{
+                   test: %OpenApiSpex.Link{
+                     operationId: "response-link-test"
+                   }
+                 },
+                 description: "Entity not found."
+               }
+             } == responses
+
+      assert %{
+               "foo" => %OpenApiSpex.Example{}
+             } = examples
+
+      assert %OpenApiSpex.Parameter{
+               description: nil,
+               name: "accept-encoding",
+               in: :header,
+               required: false,
+               allowEmptyValue: true,
+               schema: %OpenApiSpex.Schema{
+                 example: "gzip",
+                 type: :string
+               }
+             } == components_parameters_parameter
+
+      assert %{"User" => user_schema, "Admin" => admin_schema} = schemas
+
+      assert %OpenApiSpex.Schema{
+               allOf: [
+                 %OpenApiSpex.Reference{
+                   "$ref": "#/components/schemas/User"
+                 },
+                 %OpenApiSpex.Reference{
+                   "$ref": "#/components/schemas/AdminUser"
+                 }
+               ],
+               discriminator: %OpenApiSpex.Discriminator{
+                 propertyName: "userType"
+               }
+             } == admin_schema
+
+      assert %OpenApiSpex.Schema{
+               nullable: false,
+               readOnly: false,
+               writeOnly: false,
+               deprecated: false,
+               example: %{},
+               externalDocs: %OpenApiSpex.ExternalDocumentation{
+                 description: "Find more info here",
+                 url: "https://example.com"
+               },
+               properties: %{
+                 first_name: %OpenApiSpex.Schema{
+                   xml: %OpenApiSpex.Xml{
+                     namespace: "http://example.com/schema/sample",
+                     prefix: "sample"
+                   }
+                 }
+               }
+             } = user_schema
+
+      assert %OpenApiSpex.Link{
+               description: nil,
+               operationRef: nil,
+               operationId: "test",
+               requestBody: %OpenApiSpex.RequestBody{
+                 description: "link payload",
+                 content: %{
```

Comment:
`:content` should be a map with string keys: https://github.com/GhostGroup/open_api_spex/blob/daf46b32f87c24c69a16859077e6385472966f70/lib/open_api_spex/request_body.ex#L21